Source/WebCore/platform/graphics/skia/SkiaPaintingEngine.cpp

```diff
@@ -294,27 +256,35 @@ Ref<CoordinatedTileBuffer> SkiaPaintingEngine::paintLayer(const GraphicsLayer& l
 
 Ref<CoordinatedTileBuffer> SkiaPaintingEngine::postPaintingTask(const GraphicsLayer& layer, RenderingMode renderingMode, const IntRect& dirtyRect, bool contentsOpaque, float contentsScale)
 {
-    WTFBeginSignpost(this, RecordTile);
-    auto displayList = recordDisplayList(renderingMode, layer, dirtyRect, contentsOpaque, contentsScale);
-    WTFEndSignpost(this, RecordTile);
-
     auto buffer = createBuffer(renderingMode, dirtyRect.size(), contentsOpaque);
     buffer->beginPainting();
 
+    WTFBeginSignpost(this, RecordTile);
+    auto* recordingCanvas = m_pictureRecorder.beginRecording(dirtyRect.width(), dirtyRect.height());
```

Comment:
Why is the SkPictureRecorder a member? We only need it here, and after finishRecordingAsPicture it can be destroyed, no?

Reply:
Agreed. I initially thought it's too expensive to re-create it every time -- especially since it has built-in logic to reuse the fRecorder it allocated in its constructor, across recording operations (see reset()).

We can avoid the member to start with, and profile it once again and eventually optimize this, if it proofs to be expensive.